Tests/TestingApps/PurchaseTesterSwiftUI/Shared/Views/LocalReceiptView.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  LocalReceiptView.swift
+//  PurchaseTester
+//
+//  Created by Nacho Soto on 1/11/23.
+//
+
+import Foundation
+import SwiftUI
+import RevenueCat
+import ReceiptParser
+
+private typealias AppleReceipt = ReceiptParser.AppleReceipt
+
+@MainActor
+struct LocalReceiptView: View {
+
+    @State
+    private var receipt: Result<AppleReceipt, Error>?
+
+    var body: some View {
+        VStack {
+            if #available(iOS 16.0, macCatalyst 16.0, macOS 13.0, *) {
+                self.form
+                    .scrollContentBackground(.hidden)
+            } else {
+                self.form
+            }
+
+            Spacer(minLength: 10)
+
+            Button {
+                Purchases.shared.restorePurchases { _, error in
```

Comment:
Hmm I guess we need to restorePurchases before refreshing the receipt in order to get the latest receipt from apple correct? Maybe we should change the label in the button to "Refresh receipt from apple" or something like that? It might be kind of implicit so I'm ok leaving it as is as well.

Reply:
they're technically entirely separate things - restorePurchases will attach the latest purchases to the current user, but refresh receipt will refresh the receipt on device. restorePurchases may actually already refresh the receipt (depending on the status of the receipt on device)

Reply:
but with regards to the label... I think it's fine to not explicitly say "from apple" since the receipt = apple, entitlementInfo = revenuecat

Reply:
not entirely sure why we restorePurchases here, though, is it to prevent inconsistencies where the current user doesn't have the purchases on the receipt associated? I think it might be better not to have that as a side effect of the button, so you can actually see if the customerInfo doesn't have all purchases on the receipt yet

Reply:
Oh I forgot to explain this so valid confusion. We don't have any public API to refresh the receipt other than this.
#2204 introduces a way to fetch it but without refreshing it in case it's not there yet, so we need this mostly for the simulator...
I agree it doesn't make a lot of sense to call `restorePurchases` here though. Any suggestions?

Reply:
oohhhh right, that makes sense. I don't have a problem with it, but we should add a comment to explain the reasoning

Reply:
Will do 🙌🏻